Backend/spotifyRoutes.js

```diff
@@ -261,4 +238,70 @@ router.get('/spotify-user', async (req, res) => {
+
+
+        const currentUserTopArtistsData = await topArtists(currentUserSpotify)
+        const currentFriendTopArtistsData = await topArtists(currentFriendSpotify)
+        const currentUserTopArtists = currentUserTopArtistsData.items.map((artist) => [artist.name, artist.genres, artist.popularity])
+        const currentFriendTopArtists = currentFriendTopArtistsData.items.map((artist) => [artist.name, artist.genres, artist.popularity])
+
+        const currentUserFollowedArtistsData = await followedArtists(currentUserSpotify)
+        const currentFriendFollowedArtistsData = await followedArtists(currentFriendSpotify)
+        const currentUserFollowedArtists = currentUserFollowedArtistsData.artists.items.map((artist) => [artist.name, artist.genres, artist.popularity])
+        const currentFriendFollowedArtists = currentFriendFollowedArtistsData.artists.items.map((artist) => [artist.name, artist.genres, artist.popularity])
+
+        const currentUserLikedSongsData = await savedTracks(currentUserSpotify)
+        const friendUserLikedSongsData = await savedTracks(currentFriendSpotify)
+        const currentUserLikedSongs = currentUserLikedSongsData.items.map((song) => [song.track.name, [song.track.artists.map((artist) => artist.name)], song.track.popularity])
+        const friendUserLikedSongs = friendUserLikedSongsData.items.map((song) => [song.track.name, [song.track.artists.map((artist) => artist.name)], song.track.popularity])
+
+        const currentUserTopAudioFeaturesData = await audioFeatures(currentUserSpotify, currentUserTopSongsIds)
+        const currentFriendTopAudioFeaturesData = await audioFeatures(currentFriendSpotify, currentFriendTopSongsIds)
+        let currentUserDanceability = 0, currentUserLoudness = 0, currentUserEnergy = 0, currentUserTempo = 0, currentUserValence = 0;
+        let currentFriendDanceability = 0, currentFriendLoudness = 0, currentFriendEnergy = 0, currentFriendTempo = 0, currentFriendValence = 0;
+        currentUserTopAudioFeaturesData.audio_features.forEach((song) => {
+            currentUserDanceability += song.danceability
+            currentUserLoudness += song.loudness
+            currentUserEnergy += song.energy
+            currentUserTempo += song.tempo
+            currentUserValence += song.valence
+        })
+        currentFriendTopAudioFeaturesData.audio_features.forEach((song) => {
+            currentFriendDanceability += song.danceability
+            currentFriendLoudness += song.loudness
+            currentFriendEnergy += song.energy
+            currentFriendTempo += song.tempo
+            currentFriendValence += song.valence
+        })
+        const currentUserTopAudioFeatures = {danceability: parseFloat(currentUserDanceability / 10).toFixed(3), loudness: parseFloat(currentUserLoudness / 10).toFixed(3), energy: parseFloat(currentUserEnergy / 10).toFixed(3), tempo: parseFloat(currentUserTempo / 10).toFixed(3), valence: parseFloat(currentUserValence / 10).toFixed(3)}
```

Comment:
can we put this logic in a function and call it for both the current user and the friend?